modules/mini-games/Race.lua

```diff
@@ -413,11 +414,24 @@ local kill_biters = Token.register(function(name)
     end
 end)
 
+--- Unstuck yourself
+Commands.new_command('unstuck', 'Unstuck your car from walls')
+:register(function(player)
+    local car = cars[player.name]
+    if not car or not car.valid then
+        player.print("Not in a car!")
+        return Commands.error
+    end
+
+    local new_position = surface[1].find_non_colliding_position('assembling-machine-1', car.position, 5, 0.5)
+    car.teleport(new_position, surface[1])
```

Comment:
this still wont work as find_non_colliding_position does not care about rotation

Reply:
An assembling machine is 3x3.  You can drive perfectly fine when placed in the centre of a position where an assembling machine fits.

Reply:
if this would be the case the command would not be needed but as pointed out above even with a 9 x 9 shape "And even so, it's still possible to get pretty stuck driving parallel to a straight section of wall so you still need the command anyhow."

Reply:
as the command is doing the same as the car place logic

Reply:
Yes.  Drive parallel to a wall and turn and you get stuck.  You need the command to get unstuck or end up spending minutes fiddling around to get free.

Reply:
again its precisly the same as respawing the car so idk what it will do ? all it will do is try to find a non_colliding_position which crashing a car does the same (which is the problem with really getting stuck)

Reply:
Also what part of being placed in the centre of a 3x3 non-colliding area guarantees your car is not stuck is it that you don't understand?

Reply:
no what i dont get is that this is the same is respwning the cars which is the problem right?

Reply:
The cars no longer respawn inside walls.  That's what change from car to assembling-machine-1 does.

Reply:
yes so why is this command needed?

Reply:
As I told you three times now if you drive close to a straight section of a wall an turn wrong you get stuck against the wall.

Reply:
yes but youre not really stuck sutck youre not colliding so wont this just return itslef?

Reply:
i mean its called find_non_colliding_position and youre not collind so i think you wont really tp any where

Reply:
Can you fit a 3x3 tile sized assembler in the position where the car is when close up to the wall?  No!  The wall is in the way, and the car is in the way.

Why don't you test what you think is a problem and confirm that it is one _before_ you comment with speculation on your part that is simply wrong.

Reply:
o wait ofc the car is still their srry then you spam this command to get out of the walls (mabye ) also i cant test rn i am on mobile sorry

Reply:
Even if you pileup three cars around on a narrow wall like this
![image](https://user-images.githubusercontent.com/14362102/91296391-d3a5e300-e79c-11ea-8648-c48bd7330db5.png)

You're still not going to teleport through the wall with /unstuck
![image](https://user-images.githubusercontent.com/14362102/91296455-eae4d080-e79c-11ea-8902-f7af0417259c.png)

Sure, if you got like 5, 6, 7 cars around preventing any positions nearby then yeah, you'd probably be able to teleport through the wall.  But that's enough coordination to warrant the players involved being kicked for breaking the game.

Reply:
ok 👌 good srry for syaing this then i really thought it was wrong